dipy/reconst/fwdti.py

```diff
@@ -0,0 +1,608 @@
+#!/usr/bin/python
+""" Classes and functions for fitting tensors """
+from __future__ import division, print_function, absolute_import
+
+import warnings
+
+import numpy as np
+
+import scipy.optimize as opt
+
+from .base import ReconstModel
+
+from dipy.reconst.dti import (TensorFit, design_matrix, _min_positive_signal,
+                              decompose_tensor, from_lower_triangular,
+                              lower_triangular, apparent_diffusion_coef)
+from dipy.reconst.dki import _positive_evals
+
+from dipy.core.sphere import Sphere
+from .vec_val_sum import vec_val_vect
+from dipy.core.ndindex import ndindex
+
+
+def fwdti_prediction(params, gtab, Diso=3.0e-3):
+    """
+    Predict a signal given the parameters of the free water diffusion tensor
+    model.
+    Parameters
+    ----------
+    Params : ndarray
+        Model parameters. The last dimension should have the 12 tensor
+        parameters (3 eigenvalues, followed by the 3 corresponding
+        eigenvectors) the volume fraction of the free water compartment, and
+        the non diffusion-weighted signal S0
+    gtab : a GradientTable class instance
+        The gradient table for this prediction
+    Diso : float, optional
+        Value of the free water isotropic diffusion. Default is set to 3e-3
+        $mm^{2}.s^{-1}$. Please ajust this value if you are assuming different
+        units of diffusion.
+    Notes
+    -----
+    The predicted signal is given by:
+    $S(\theta, b) = S_0 * [(1-f) * e^{-b ADC} + f * e^{-b D_{iso}]$, where
+    $ADC = \theta Q \theta^T$, $\theta$ is a unit vector pointing at any
+    direction on the sphere for which a signal is to be predicted, $b$ is the b
+    value provided in the GradientTable input for that direction, $Q$ is the
+    quadratic form of the tensor determined by the input parameters, $f$ is the
+    free water diffusion compartment, $D_{iso}$ is the free water diffusivity
+    which is equal to $3 * 10^{-3} mm^{2}s^{-1} [1]_.
+    References
+    ----------
+    .. [1] Pasternak, O., Sochen, N., Gur, Y., Intrator, N., Assaf, Y., 2009.
```

Comment:
I don't think this is the right reference here, considering this is not an implementation of the Pasternak algorithm.